src/components/App/App.jsx

```diff
@@ -1,115 +1,210 @@
+import VendorProfile from '../Profile/VendorProfile';
+import RegisterVendorPage from '../RegisterPage/RegisterVendorPage';
 
 import './App.css';
 
+// Material-UI:
+import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';
+import { CssBaseline } from '@material-ui/core';
+
+const contriveTheme = createMuiTheme({
+  palette: {
+    background: {
+      paper: '#fff',
+      default: '#fff',
+      level2: '#f5f5f5',
+      level1: '#fff',
+    },
+    secondary: {
+      light: '#fffff9',
+      main: '#fefcc6',
+      dark: '#cac995',
+      contrastText: '#000000',
+    },
+    primary: {
+      light: '#cd7c50',
+      main: '#984f26',
+      dark: '#652500',
+      contrastText: '#ffffff',
+    },
+  },
+  typography: {
+    fontFamily: ['Forum', 'Raleway'].join(','),
+  },
+});
+
 function App() {
   const dispatch = useDispatch();
 
   useEffect(() => {
     dispatch({ type: 'FETCH_USER' });
+    dispatch({ type: 'FETCH_VENDOR_TYPES' });
+    dispatch({ type: 'FETCH_SPECIAL_FEATURES' });
   }, [dispatch]);
 
   return (
-    <Router>
-      <div>
+    <ThemeProvider theme={contriveTheme}>
+      <CssBaseline />
+      <Router>
         <Nav />
-        <Switch>
-          {/* Visiting localhost:3000 will redirect to localhost:3000/home */}
-          <Redirect exact from="/" to="/home" />
-
-          {/* Visiting localhost:3000/about will show the about page. */}
-          <Route
-            // shows AboutPage at all times (logged in or not)
-            exact
-            path="/about"
-          >
-            <AboutPage />
-          </Route>
-
-          {/* For protected routes, the view could show one of several things on the same route.
+        <div>
+          <Switch>
+<<<<<<< HEAD
+            {/* Visiting localhost:3000 will redirect to localhost:3000/home */}
+
+            <Redirect exact from="/" to="/home" />
+
+=======
+            {/* Visiting localhost:3000 will redirect to localhost:3000/welcome, 
+            which redirects to localhost:3000/discover if user is already logged in (see below) */}
+            <Redirect exact from="/" to="/welcome" />
+>>>>>>> 12df79ba2130e1d78a329ac307175a6943af0c6f
+
+            {/* Visiting localhost:3000/about will show the about page. */}
+            <Route
+              // shows AboutPage at all times (logged in or not)
+              exact
+              path="/about"
+            >
+              <AboutPage />
+            </Route>
+
+            {/* Visiting localhost:3000/styleGuide will show the styleGuide used for this page.
+            This route should be removed/inaccessible by users for production.*/}
+            <Route
+              // shows the StyleGuide for this Project
+              exact
+              path="/styleGuide"
+            >
+              <StyleGuide />
+            </Route>
+
+            <Route exact path="/messages">
+              <MessagesList />
+            </Route>
+
+            <Route exact path="/message/:id">
```

Comment:
**CHANGE REQUEST**
Should be protected route

Reply:
Completed - John Shands